src/devices/deltadore_x3d.c

```diff
@@ -0,0 +1,191 @@
+- Unknown      {8}  Influenced by header len
+   - 0x00        on header len 0x0c
+   - 0x08        on header len 0x0f
+     - unknown      {8} 0x00
+     - temperature {16} little-endian multiplied by 100 -> 2050 = 20.5 °C
+- Msg Id      {32}
+- Retry Cnt    {8} used for msg retry ? and direction
+   - lower nibble and zero send from thermostat, count downwards
+   - upper nibble send from switch actor, count upwards
+- Act No.      {8} target actor number
+payload data
+- CRC16       {16}
+
+The length including payload is whitened using CCITT whitening enabled in SX1211 chipset.
+The payload contains some garbage at the end. The documentation of the SX1211 assume to
+exclude the length byte from length calculation, but the CRC16 checksum at the end is so placed,
+that the length byte is included. Maybe someone read the docs wrong. The garbage after the
+checksum contains data from previous larger messages.
+
+So the last two bytes contains the CRC16(Poly=0x1021,Init=0x0000) value.
+
+To get raw data:
+
+    ./rtl_433 -f 868.95M -X 'n=DeltaDore,m=FSK_PCM,s=25,l=25,r=25600,preamble=aa8169967e'
+*/
+
+#include "decoder.h"
+
+static void ccitt_dewhitening(uint8_t *whitening_key_msb_p, uint8_t *whitening_key_lsb_p, uint8_t *buffer, uint16_t buffer_size)
+{
+    uint8_t whitening_key_msb = *whitening_key_msb_p;
+    uint8_t whitening_key_lsb = *whitening_key_lsb_p;
+    uint8_t whitening_key_msb_previous;
+    uint8_t reverted_whitening_key_lsb = whitening_key_lsb;
+
+    for (uint16_t buffer_pos = 0; buffer_pos < buffer_size; buffer_pos++) {
+
+        reverted_whitening_key_lsb = (whitening_key_lsb & 0xf0) >> 4 | (whitening_key_lsb & 0x0f) << 4;
+        reverted_whitening_key_lsb = (reverted_whitening_key_lsb & 0xcc) >> 2 | (reverted_whitening_key_lsb & 0x33) << 2;
+        reverted_whitening_key_lsb = (reverted_whitening_key_lsb & 0xaa) >> 1 | (reverted_whitening_key_lsb & 0x55) << 1;
+
+        buffer[buffer_pos] ^= reverted_whitening_key_lsb;
+
+        for (uint8_t rol_counter = 0; rol_counter < 8; rol_counter++) {
+            whitening_key_msb_previous = whitening_key_msb;
+            whitening_key_msb = (whitening_key_lsb & 0x01) ^ ((whitening_key_lsb >> 5) & 0x01);
+            whitening_key_lsb = ((whitening_key_msb_previous << 7) & 0x80) | ((whitening_key_lsb >> 1) & 0xff);
+        }
+    }
+
+    *whitening_key_msb_p = whitening_key_msb;
+    *whitening_key_lsb_p = whitening_key_lsb;
+}
+
+static int deltadore_x3d_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    uint8_t const preamble[] = {
+            /*0xaa, 0xaa, */ 0xaa, 0xaa, // preamble
+            0x81, 0x69, 0x96, 0x7e       // sync word
+    };
+
+    data_t *data;
+
+    if (bitbuffer->num_rows != 1) {
+        return DECODE_ABORT_EARLY;
+    }
+
+    int row = 0;
+    // Validate message and reject it as fast as possible : check for preamble
+    unsigned start_pos = bitbuffer_search(bitbuffer, row, 0, preamble, sizeof (preamble) * 8);
```

Comment:
maybe add `+ sizeof (preamble) * 8)` to set start_pos to first data byte and clean up all subsequent references. Also `>=` in the next `if`